### DISCUSSION TITLE
Rule Tile / Auto Tiles

### DISCUSSION THREAD

**Discussion by frederickjjoubert** [StarArawn/bevy_ecs_tilemap#323] [Ideas]
Hello, I hope to start a discussion on working with Rule Tiles / Auto Tiles with the `bevy_ecs_tilemap` plugin. 

Is this something that could be added in the future so that a tile's texture would change depending on the conditions of neighboring tiles?

Do you have any recommendations to add this on top of `bevy_ecs_tilemap` if this isn't planned or won't be added for a long time?

**Comment by StarArawn** [comment]
> Hello, I hope to start a discussion on working with Rule Tiles / Auto Tiles with the `bevy_ecs_tilemap` plugin.
> 
> Is this something that could be added in the future so that a tile's texture would change depending on the conditions of neighboring tiles?
> 
> Do you have any recommendations to add this on top of `bevy_ecs_tilemap` if this isn't planned or won't be added for a long time?

There are no plans currently to add rule or auto tiles. I think my biggest gripe with them is the many differing solutions to the problem and a lack of a user friendly API. Tiled for example lets you draw rules but it takes forever to setup right and is more frustrating. Especially because tileset X might use another method.

I would be open to adding something to bevy_ecs_tilemap, I'm just not sure what that would look like yet. Maybe an RFC process would work here for a complex feature like this?

**Reply by frederickjjoubert** [reply]
> > Hello, I hope to start a discussion on working with Rule Tiles / Auto Tiles with the `bevy_ecs_tilemap` plugin.
> > Is this something that could be added in the future so that a tile's texture would change depending on the conditions of neighboring tiles?
> > Do you have any recommendations to add this on top of `bevy_ecs_tilemap` if this isn't planned or won't be added for a long time?
> 
> There are no plans currently to add rule or auto tiles. I think my biggest gripe with them is the many differing solutions to the problem and a lack of a user friendly API. Tiled for example lets you draw rules but it takes forever to setup right and is more frustrating. Especially because tileset X might use another method.
> 

I have used the RuleTiles provided by Unity before, and I think they offer a good solution for addressing the different methods rule tiles can take. The API is quite straightforward. You create the "rules" for which neighboring positions are allowed, not allowed, or don't matter, and which sprite should then be selected. Setup is quite tedious, but once set up it works very well.

Unity RuleTiles
- [RuleTile](https://docs.unity3d.com/Packages/com.unity.2d.tilemap.extras@3.0/manual/RuleTile.html)
- [Using Rule Tiles](https://learn.unity.com/tutorial/using-rule-tiles#5fe9914fedbc2a28d93ce45d)

Methods / Layouts
- [Blob, Sub-blob, micro-blob](https://www.boristhebrave.com/2013/07/14/tileset-roundup/?q=tutorials/tileset-roundup)
- [Wang](http://www.cr31.co.uk/stagecast/wang/intro.html)
- [Others](https://www.tilesetter.org/docs/tileset_behavior)

> I would be open to adding something to bevy_ecs_tilemap, I'm just not sure what that would look like yet. Maybe an RFC process would work here for a complex feature like this?

I would be happy to start this.

**Comment by frederickjjoubert** [comment]
Let us continue this discussion here: https://github.com/StarArawn/bevy_ecs_tilemap/discussions/326
